app/src/components/features/rules/SharedListRuleViewerModal/index.js

```diff
@@ -2,20 +2,26 @@ import React from "react";
 import { Modal } from "antd";
 import RuleBuilder from "../RuleBuilder";
 
-const SharedListViewerModal = ({ rule, isOpen, toggle }) => {
+const SharedListViewerModal = ({ rule, isOpen, toggle, previewType }) => {
   if (!rule) {
     return <span></span>;
   }
   return (
     <Modal
       className="modal-dialog-centered max-width-80-percent "
-      visible={isOpen}
+      open={isOpen}
       onCancel={toggle}
       footer={null}
       title={`${rule.name} (Preview Mode)`}
       width="80%"
     >
-      <RuleBuilder rule={rule} isSharedListViewRule={true} />
+      <RuleBuilder
+        rule={rule}
+        isSharedListViewRule={true}
+        source={
+          previewType === "trash" ? "trash_preview" : "shared_list_preview"
```

Comment:
Only the rule preview is shown with this modal. The rule editor does not open even when sharedlist is imported. Do we want this event this? Please confirm this